src/Microsoft.Extensions.Configuration.Abstractions/ConfigurationExtensions.cs

```diff
@@ -26,17 +26,28 @@ public static string GetConnectionString(this IConfiguration configuration, stri
         /// </summary>
         /// <param name="configuration">The <see cref="IConfiguration"/> to enumerate.</param>
         /// <returns>An enumeration of key value pairs.</returns>
-        public static IEnumerable<KeyValuePair<string, string>> AsEnumerable(this IConfiguration configuration)
+        public static IEnumerable<KeyValuePair<string, string>> AsEnumerable(this IConfiguration configuration) => configuration.AsEnumerable(removePathFromChildKeys: false);
+
+        /// <summary>
+        /// Get the enumeration of key value pairs within the <see cref="IConfiguration" />
+        /// </summary>
+        /// <param name="configuration">The <see cref="IConfiguration"/> to enumerate.</param>
+        /// <param name="removePathFromChildKeys">If true, removes the configuration path from the child keys.</param>
```

Comment:
If true, the child keys returned will have the current configuration's Path trimmed from the front.